### PR TITLE
Add rust to CI

### PR DESCRIPTION
**Pull request by erikjohnston**
We only run the CI when the rust directory has changed.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -65,10 +82,54 @@ jobs:
...
-    needs: [lint, lint-crlf, lint-newsfile, lint-pydantic, check-sampleconfig, check-schema-delta]
+    needs:
+      - lint
+      - lint-crlf
+      - lint-newsfile
+      - lint-pydantic
+      - check-sampleconfig
+      - check-schema-delta
+      - lint-clippy
+      - lint-rustfmt
```
Comment (DMRobertson) [inline .github/workflows/tests.yml:132]: I don't know how many rust CI jobs we expect to have. If runner contention is an issue we could merge these... but it's probably premature to do so now.

**Review by DMRobertson — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (erikjohnston) [reply .github/workflows/tests.yml:132]: Yup, my thoughts too. Plus these only run on PRs that change rust code, so I doubt they'll even be run that frequently.

**Comment by DMRobertson** [comment]
Erm, this run on develop is judged to have failed... but everything passed?

https://github.com/matrix-org/synapse/runs/8269539943?check_suite_focus=true

**Comment by DMRobertson** [comment]
Oh I think we just need to add `cargo-test` to `skippable`

**Comment by erikjohnston** [comment]
Woops, fixed in #13769

**Pull request merged**